File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.*;
class Solution {
    public int solution(int[] elements) {
        int answer = 0;
        HashSet<Integer> set = new HashSet<>();
        int[] eleDouble = new int[elements.length * 2];
        for (int i = 0; i < eleDouble.length; i++) {
            eleDouble[i] = elements[i % elements.length];
        }
        for (int i = 0; i < elements.length; i++) {
            for (int j = 0; j < elements.length; j++) {
                set.add(Arrays.stream(eleDouble, j, j+i).sum());
            }
        }
        return set.size();
    }
}

File: 프로그래머스/lv2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    answer = 0
    idx = 1
    for i in range(len(elements)):
        temp = so(idx, elements)
        answer += temp
        print(answer)
        idx += 1
    return answer


def so(n, arr):
    arr_num = []
    for start in range(-1, len(arr) - 1):
        cnt = 0
        j = start
        for j in range(n):
            cnt += arr[j]
            print(cnt)
            j += 1
        if cnt not in arr_num:
            arr_num.append(cnt)
    return len(arr_num)